### DISCUSSION TITLE
Extra map details about patterns in UI

### DISCUSSION THREAD

**Discussion by Rekunan** [ppy/osu#27262] [Ideas]
SR and object counts seem to be the only shown stat effective at communicating ahead of time about the patterns inside a map, which can be very broad and lacking at times, especially if you're a new player and are unaware about typical mapper/song styles. Somewhat borrowing an idea from VSRGs would be showing more specific details about the patterns in the map.

2 that come to mind are:
- Peak velocity
- Longest stream (with bpm and how often)

`Peak velocity` would be just the pattern that has the largest `osu!px`/`time`.
To get the `longest stream`, you could find the average bpm of the note placements on the map, and then count anything above that as a stream, at which point you'd count the longest "stream" of notes. If there's nothing above, then just use the average as the count. Bpm would be dividing the calculated bpm by 2, since basically everyone plays streams with 2 fingers. How often is just counting how many times a pattern with that many consecutive notes for that bpm show up

Some hand-picked examples of the 2nd detail with possible benefits:
[Apollo - Brazil](https://osu.ppy.sh/beatmapsets/2104327#osu/4415495): `Longest stream: 25 notes (111.5bpm) (1 time)`
This tells me that the map is either slow, or has nothing faster than jumps, and that there is 1 long consecutive jump section.
[Vektor - Fast Paced Society](https://osu.ppy.sh/beatmapsets/298996#osu/670881): `Longest stream: 281 notes (208bpm) (1 time)`
This tells me that the map has a really long deathstream, and therefore, probably has smaller streams throughout the map.
[Fujijo Seitokai Shikkou-bu - Best FriendS](https://osu.ppy.sh/beatmapsets/249939#osu/574471): `Longest stream: 5 notes (163bpm) (4 times)`
This tells me that the map is either jump or burst focused, and is more likely to be the former.
[HujuniseikouyuuP - Talent Shredder](https://osu.ppy.sh/beatmapsets/47710#osu/153857): `Longest stream: 7 notes (140bpm) (7 times)`
This tells me that the map is either jump or burst focused, and is more likely to be the latter.

**Comment by Givikap120** [comment]
I think it's better and just more general to use Difficulty Attributes
Aim + Speed difficulties + BPM is usually enough to guess the map

**Reply by Rekunan** [reply]
I don't see why we can't have both; the combination of this + difficulty attributes would become extremely useful, essentially removing the doubts from the stats that I mentioned in my examples.
